scripts/check-index.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+
+# Get the past 4 Kubernetes releases not counting the latest available
+# example: 1.19 (latest), 4 last releases: 1.18, 1.17, 1.16, 1.15
+LASTRELEASES=4
+COUNTER=0
+RELEASES=()
+INDEXFILE=dist/index.html
+EXITSTATUS=0
+
+LATEST=$(curl -Ls https://dl.k8s.io/release/stable.txt)
+LATESTMAJOR=`echo $LATEST | cut -dv -d. -f1`
+LATESTMINOR=`echo $LATEST | cut -d. -f2`
+RELEASES+=($LATEST)
+
+while [  $COUNTER -lt $LASTRELEASES ]; do
+  let "LATESTMINOR-=1"
+  TEMP=$(curl -Ls https://dl.k8s.io/release/stable-${LATESTMAJOR:1}.$LATESTMINOR.txt)
+  RELEASES+=("$TEMP")
+  let COUNTER+=1
+done
+
+echo ${RELEASES[@]}
+
+echo "Checking if index.html is up to date"
+for tag in "${RELEASES[@]}"
+do
+  case `grep -Rq "$tag" "$INDEXFILE" >/dev/null; echo $?` in
+    0)
+      echo "$tag found in the index.html"
+      EXITSTATUS=0
```

Comment:
We override the exist status is if a check failed in a previous loop iteration, which I guess is not intended:
```
> ./scripts/check-index.sh
v1.19.0 v1.18.8 v1.17.11 v1.16.15 v1.15.12
Checking if index.html is up to date
v1.19.0 not found in the index.html, please update the index.html file
v1.18.8 found in the index.html
v1.17.11 found in the index.html
v1.16.15 not found in the index.html, please update the index.html file
v1.15.12 found in the index.html

> echo $status
0
```

Reply:
Hmmm, I think I'd prefer to not play with exit statuses in conditionals.
I'd suggest encapsulating the check in a function:

```shell
func stuff() {
  <do-stuff>
  if <bad-thing-happened>; then
    return 1
  fi
}

for foo in foos; do
  stuff || exit <1 or 0>
done
```

Reply:
yes, was not the idea :/

will fix and follow the Stephen's idea

Reply:
with the new update we get this:
```shell
$ ./scripts/check-index.sh
Getting Kubernetes releases
Will validate the index.html using the following releases: v1.19.0 v1.18.8 v1.17.11 v1.16.15 v1.15.12
v1.19.0 not found in the index.html, please update the index.html file
v1.16.15 not found in the index.html, please update the index.html file

$ echo $?
1
```